crates/rolldown_binding/src/parallel_js_plugin_registry.rs

```diff
@@ -0,0 +1,96 @@
+use crate::options::plugin::{BindingPluginOptions, BindingPluginWithIndex};
+use napi::{
+  bindgen_prelude::{FromNapiValue, Object, ObjectFinalize},
+  Env, JsUnknown,
+};
+use napi_derive::napi;
+use once_cell::sync::Lazy;
+use rustc_hash::FxHashMap;
+use std::{
+  hash::BuildHasherDefault,
+  sync::{
+    atomic::{self, AtomicU16},
+    Mutex,
+  },
+};
+
+type PluginsInSingleWorker = Vec<BindingPluginWithIndex>;
+type PluginsList = Vec<PluginsInSingleWorker>;
+pub(crate) type PluginValues = FxHashMap<usize, Vec<BindingPluginOptions>>;
+
+static PLUGINS_MAP: Lazy<Mutex<FxHashMap<u16, PluginsList>>> = Lazy::new(Mutex::default);
+static NEXT_ID: AtomicU16 = AtomicU16::new(1);
+
+#[napi(custom_finalize)]
```

Comment:
Why `custom_finalize`? I thought `ParallelJsPluginRegistry` was just a simple object,

Reply:
If an error is thrown between https://github.com/rolldown/rolldown/blob/568038e3f0e92cd37a6eb0b3f603451bc5ee8f7b/packages/rolldown/src/utils/create-bundler.ts#L20-L22 and https://github.com/rolldown/rolldown/blob/568038e3f0e92cd37a6eb0b3f603451bc5ee8f7b/packages/rolldown/src/utils/create-bundler.ts#L33-L37 , `ParallelJsPluginRegistry` will be dropped when it's GCed on JS side, but the values in `PLUGINS_MAP` won't be dropped. I think we need to manually clear them to avoid memory leak.
https://github.com/rolldown/rolldown/blob/568038e3f0e92cd37a6eb0b3f603451bc5ee8f7b/crates/rolldown_binding/src/parallel_js_plugin_registry.rs#L65-L70

Reply:
I guess that's I want to have a `BundlerId` to link all resources to a bundler and drop everything while something is wrong. However, I also can't be sure how this really works out. Let's use the current implementation for now.